addons/mail/static/src/core/web/recipients_input.js

```diff
@@ -150,18 +150,27 @@ export class RecipientsInput extends Component {
 
                     if (isEmail(email)) {
                         createOption.onSelectOption = async () => {
-                            const partners = await rpc("/mail/partner/from_email", {
-                                thread_model: this.props.thread.model,
-                                thread_id: this.props.thread.id,
-                                emails: [email],
-                            });
+                            const partners =
+                                email === name
+                                    ? await rpc("/mail/partner/from_email", {
+                                          thread_model: this.props.thread.model,
+                                          thread_id: this.props.thread.id,
+                                          emails: [email],
+                                      })
+                                    : await this.orm.create("res.partner", [
```

Comment:
Seems very louche. As name and email come from term, can't you use term in from_email route ?